src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -201,6 +208,29 @@ public FileResource(FileService fileService, ResourceLoader resourceLoader, User
         return responseEntityForFilePath(Constants.DRAG_ITEM_FILEPATH + filename);
     }
 
+    /**
+     * GET /files/file-upload/submission/:submissionId/:filename : Get the file upload exercise submission file
+     *
+     * @param submissionId ID of the submission, the attachment belongs to
+     * @param filename  the filename of the file
+     * @param temporaryAccessToken The access token is required to authenticate the user that accesses it
+     * @return The requested file, 403 if the logged in user is not allowed to access it, or 404 if the file doesn't exist
+     */
+    @GetMapping("/files/file-upload-submission/{submissionId}/{filename:.+}")
+    @PreAuthorize("permitAll()")
+    public ResponseEntity getFileUploadSubmission(@PathVariable Long submissionId, @PathVariable String filename, @RequestParam("access_token") String temporaryAccessToken) {
+        log.debug("REST request to get file : {}", filename);
+        Optional<FileUploadSubmission> optionalSubmission = fileUploadSubmissionRepository.findById(submissionId);
```

Comment:
I am not sure what can be refactored here, because the submission that I get is specific to File Upload, also later I have temporary access token validaton and I want to show custom message for them.